pyomo/core/base/set.py

```diff
@@ -1386,6 +1695,9 @@ def _sort(self):
     'set_value', 'add', 'remove', 'discard', 'clear', 'update', 'pop',
 )
 
+
+@ModelComponentFactory.register(
+    "Set data that is used to define a model instance.")
```

Comment:
This is the first I have seen or heard of the ModelComponentFactory. Is this new? Can you explain the purpose of this?

Reply:
This is ancient (pre-Coopr3).  It is only used by a legacy subcommand that will print out the list of known components and the `@model.Constraint()` decorator syntax.